docs/resources/control-flow/user-interactivity/forms/form-validation.md

```diff
@@ -0,0 +1,191 @@
+---
+slug: form-validation
+title: Form Validation
+tags: []
+description: Learn how to add Form Validation widget in your FlutterFlow app.
+sidebar_position: 2
+---
+
+
+# Form Validation
+
+You can add validations to input fields by wrapping them inside the Form widget. The Form widget enables you to validate user inputs and display appropriate messages when validation criteria are not met.
+
+For example, you could use it to check if a user has given a valid email and password. This makes it easy to handle user input and ensure that the data is correct before it is submitted to the server or stored locally.
+
+## Adding Form widget
+
+Let's see how to add a *Form* widget by building a signup example. Here's how it looks:
+
+![Validating input fields in a signup form](../imgs/form-demo.gif)
+
+Building and validating a *Form* includes the following steps:
+
+1. [Adding input fields](#1-adding-input-fields)
+2. [Adding validations](#2-adding-validations)
+3. [Adding validate action](#3-adding-validate-action)
+
+### 1. Adding input fields
+
+A form widget can only validate if there are any input fields. Here's an example of adding input fields for the signup form.
+
+1. First, add the **Form** widget itself from the **Form Elements**.
+2. Inside the form, add the [**Column**](#) widget from the **Layout Elements** tab.
+3. Now, add two [**TextFields**](form-widgets/textfield) (one for email and one for password).
+4. Add a [**Button**](#) widget and then add [Date/Time Picker](#) action to get the date of birth.
+5. Add one more [**Button**](#) to validate and submit the form.
+
+Here's how it looks:
+
+![Input fields](../imgs/fv-input-fields.avif)
+
+### 2. Adding validations
+
+Validation refers to the process of checking user input for correctness and ensuring that it meets certain criteria or requirements. This can include checking for the presence of required fields, verifying that a value is within a certain range or format, or validating against the custom pattern.
+
+After adding input fields, they will be available to be validated using the form widget properties. Here's how you do it:
+
+1. Select the **Form** widget, and move to the **Properties Panel > Validate** section.
+2. Identify the **TextField** on which you would like to add the validation and tick the box on the right side.
+    1. Inside the **Error Message** input box, provide the message that will be displayed (below the *TextField*) if a user leaves the *TextField* empty.
+    2. You can also specify the **Min Required Character** and **Max Allowed Characters**.
+        1. **Min Required Character**: This is the minimum character required for the validation to pass. For example, If you provide a value as 9 and a user enters the value as *a@a.com* (which is 6 characters), **then the validation fails, and an error message will be displayed.
```

Comment:
In the future or in your review fix (optional), we should split these into smaller headers, because it is not very easy to follow these many levels of nesting. (applies to many old PR migrations that we should fix in the future)

Reply:
Ya, no worries. we'll take care of this while rewriting 👍